lte/gateway/c/oai/common/common_defs.h

```diff
@@ -105,15 +106,18 @@ typedef enum {
   sIZE += sizeof(uint8_t)
 
 #define DECODE_U16(bUFFER, vALUE, sIZE)                                        \
-  vALUE = ntohs(*(uint16_t*) (bUFFER));                                        \
+  memcpy((unsigned char*) &vALUE, bUFFER, sizeof(uint16_t));                   \
+  vALUE = ntohs(vALUE);                                                        \
   sIZE += sizeof(uint16_t)
 
 #define DECODE_U24(bUFFER, vALUE, sIZE)                                        \
-  vALUE = ntohl(*(uint32_t*) (bUFFER)) >> 8;                                   \
+  memcpy((unsigned char*) &vALUE, bUFFER, sizeof(uint32_t));                   \
+  vALUE = ntohl(vALUE) >> 8;                                                   \
```

Comment:
This doesn't look correct to me if network byte order and host byte order are different. E.g.,
buffer -> byte-1, byte-2, byte-3, byte-4, byte-5, ....
Line 114: vALUE -> byte-1, byte-2, byte-3, byte-4
Line 115: nothl(vALUE) -> byte-4, byte-3, byte-2, byte-1  
                ntohl(vALUE) >> 8 -> byte-4, byte-3, byte-2

instead the target was to have byte-3, byte-2, byte-1. Do I see this wrong?